stonesoup/hypothesiser/gaussianmixture.py

```diff
@@ -93,13 +93,25 @@ def hypothesise(self, components, detections, timestamp):
             # Get miss detected components
             miss_detections_hypothesis = MultipleHypothesis(
                 [x for x in single_hypothesis_list if not x])
+            
+            # Get Soft detections
+            soft_detect_list = [x for x in single_hypothesis_list if x]
             for detection in detections:
-                # Create multiple hypothesis per detection
-                detection_multiple_hypothesis = \
-                    MultipleHypothesis(list([hypothesis for hypothesis in single_hypothesis_list
+                if hasattr(detection, 'components') == True:      
+                    for sub_detection in detection.components:
+                        # Create multiple hypothesis per GM component
+                        detection_multiple_hypothesis = \
+                            MultipleHypothesis(list([hypothesis for hypothesis in soft_detect_list
+                                                if hypothesis.measurement.components == sub_detection])) 
```

Comment:
You are correct: sub_detection is of type (Tagged)WeightedGaussianState, while hypothesis.measurement.components is of type [(Tagged)WeightedGaussianState], i.e. a list. I ran code many times to check if this is an appropriate comparision or not. It seems it is OK. 

However, to fully address your concern, I have rewriten the comparision as follows:
 if hypothesis.measurement.components[0] == sub_detection since [(Tagged)WeightedGaussianState] has only one element.